homeassistant/components/wallbox/coordinator.py

```diff
@@ -133,10 +151,10 @@ async def _async_update_data(self) -> dict[str, Any]:
         """Get new sensor data for Wallbox component."""
         return await self.hass.async_add_executor_job(self._get_data)
 
+    @_require_authentication
     def _set_charging_current(self, charging_current: float) -> None:
         """Set maximum charging current for Wallbox."""
         try:
-            self._authenticate()
             self._wallbox.setMaxChargingCurrent(self._station, charging_current)
         except requests.exceptions.HTTPError as wallbox_connection_error:
```

Comment:
Try and the following try/except can be removed als the decorator handles the same errors

Reply:
yes, but also no, the decorator only handles these exceptions for the auth endpoint -> they can also occur for the other API endpoints. The authorization endpoint can be available but another one can't, also, you can get a FORBIDDEN if you try to get data from a wallbox that isn't yours.

Reply:
yes, I removed them for most functions that are decorated except the _lock_unlock and _set_charging_current. For these functions I want to catch the possibility that someone has a limited account and does not have permission to set these functions. In this case [ Can login] [ Cannot Lock / Change current] I raise InvalidAuth which I use in the number / lock component to not load these parts of the integration.